src/core/ucc_mc.h

```diff
@@ -103,17 +105,38 @@ ucc_status_t ucc_mc_reduce_multi_alpha(
     size_t stride, ucc_datatype_t dtype, ucc_reduction_op_t reduce_op,
     ucc_reduction_op_t vector_op, double alpha, ucc_memory_type_t mem_type);
 
-static inline ucc_status_t ucc_dt_reduce(const void *src1, const void *src2,
+static inline ucc_status_t ucc_mc_reduce_userdefined(void *src1, void *src2,
+                                                     void *dst, size_t n_vectors,
+                                                     size_t count, size_t stride,
+                                                     ucc_dt_generic_t *dt)
+{
+    ucc_reduce_cb_params_t params = {
+        .src1      = src1,
+        .src2      = src2,
+        .dst       = dst,
+        .n_vectors = n_vectors,
+        .count     = count,
+        .stride    = stride,
+        .dt        = dt
+    };
+
+    dt->ops.reduce.cb(&params);
+    return UCC_OK;
+}
+
+static inline ucc_status_t ucc_dt_reduce(void *src1, void *src2,
```

Comment:
why "const" is removed for src1 , src2?

Reply:
we don't have it for reduce multi neither, the reason is: src2 and dst can actually overlap (however in a very special way that still allows dst[j] = sum (src + stride * i ) [j], the dst can be equal to src2 , e.g. - it will be "inplace" reduce). This is why don't have const in reduce_cb as well.